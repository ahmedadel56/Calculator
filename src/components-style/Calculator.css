@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@1,300&display=swap");

* {
  font-family: "Roboto", sans-serif;
}

.container {
  display: flex;
  flex-direction: column;
  height: 30rem;
  width: 30rem;
  margin: auto;
  margin-top: 3rem;
  background-color: #e0e0e0;
}

.output {
  height: 11%;
  background-color: #858693;
  display: flex;
  align-items: center;
  justify-content: end;
  padding-right: 10px;
  color: #e0e0e0;
  font-size: 25px;
}

.buttons {
  height: 85%;
  display: grid;
  grid-template-columns: 24.5% 24.5% 24.5% 24%;
  grid-template-rows: 20% 20% 20% 20% 20%;
  grid-gap: 0.26rem;
}

button {
  border: none;
  outline: none;
  font-size: 22px;
}

button:active {
  opacity: 0.5;
}

button:nth-of-type(4),
button:nth-of-type(8),
button:nth-of-type(12),
button:nth-of-type(16),
button:nth-of-type(19) {
  background-color: #f5913e;
}

.not-equal {
  grid-column: 1 / span 2;
}
