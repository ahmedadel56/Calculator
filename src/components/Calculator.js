import React, { useState } from 'react';
import '../components-style/Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const numbersHandler = (e) => {
    const { total, next, operation } = state;

    if (!total && !operation) {
      setState({ ...state, total: e.target.innerHTML });
    } else if (total && !operation) {
      setState({ ...state, total: total + e.target.innerHTML });
    } else if (total && operation && !next) {
      setState({ ...state, next: e.target.innerHTML });
    } else {
      setState({ ...state, next: next + e.target.innerHTML });
    }
  };

  const operationHandler = (e) => {
    setState({ ...state, operation: e.target.innerHTML });
  };

  const handleResults = (e) => {
    const calculation = calculate(state, e.target.innerHTML);

    setState(calculation);
  };

  const { total, next, operation } = state;

  return (
    <div className="container" data-testid="Calculator">
      <div className="output">
        {total || 0}
        {operation}
        {next}
      </div>
      <div className="buttons">
        <button type="button" onClick={handleResults}>
          AC
        </button>
        <button type="button" onClick={handleResults}>
          +/-
        </button>
        <button type="button" onClick={operationHandler}>
          %
        </button>
        <button type="button" onClick={operationHandler}>
          รท
        </button>
        <button type="button" onClick={numbersHandler}>
          7
        </button>
        <button type="button" onClick={numbersHandler}>
          8
        </button>
        <button type="button" onClick={numbersHandler}>
          9
        </button>
        <button type="button" operation="x" onClick={operationHandler}>
          x
        </button>
        <button type="button" onClick={numbersHandler}>
          4
        </button>
        <button type="button" onClick={numbersHandler}>
          5
        </button>
        <button type="button" onClick={numbersHandler}>
          6
        </button>
        <button type="button" operation="-" onClick={operationHandler}>
          -
        </button>
        <button type="button" onClick={numbersHandler}>
          1
        </button>
        <button type="button" onClick={numbersHandler}>
          2
        </button>
        <button type="button" onClick={numbersHandler}>
          3
        </button>
        <button type="button" operation="+" onClick={operationHandler}>
          +
        </button>
        <button
          type="button"
          buttonName="0"
          className="not-equal"
          onClick={numbersHandler}
        >
          0
        </button>
        <button type="button" buttonName="." onClick={handleResults}>
          .
        </button>
        <button type="button" buttonName="=" onClick={handleResults}>
          =
        </button>
      </div>
    </div>
  );
};
export default Calculator;
