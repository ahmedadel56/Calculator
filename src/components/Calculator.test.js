import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('<Calculator />', () => {
  test('it should mount', () => {
    render(<Calculator />);

    const calculator = screen.getByTestId('Calculator');

    expect(calculator).toBeInTheDocument();
  });

  it('it should match snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
