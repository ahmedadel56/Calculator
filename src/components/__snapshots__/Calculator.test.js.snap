// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> it should match snapshot 1`] = `
<div
  className="container"
  data-testid="Calculator"
>
  <div
    className="output"
  >
    0
  </div>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      operation="x"
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      operation="-"
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      operation="+"
      type="button"
    >
      +
    </button>
    <button
      buttonName="0"
      className="not-equal"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      buttonName="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      buttonName="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
